Zend/zend_execute.c

```diff
@@ -1293,15 +1440,44 @@ static zend_never_inline void zend_pre_incdec_overloaded_property(zval *object,
 		}
 		ZVAL_DEREF(z);
 		SEPARATE_ZVAL_NOREF(z);
-		if (inc) {
-			increment_function(z);
+		if (ZEND_OBJECT_HAS_TYPE_HINTS(object) && (prop_info = zend_object_fetch_property_type_info(object, property, cache_slot))) {
+			zval tmp;
+
+			ZVAL_DUP(&tmp, z);
```

Comment:
> we even don't exactly know if this property has a type hint.

We do, we got prop info for it ... that seems like the correct way to do it
?

Cheers
Joe

On Wed, Apr 6, 2016 at 11:48 AM, Dmitry Stogov <notifications@github.com>
wrote:

> In Zend/zend_execute.c
> <https://github.com/php/php-src/pull/1797#discussion_r58685401>:
>
> > @@ -1293,15 +1440,44 @@ static zend_never_inline void zend_pre_incdec_overloaded_property(zval *object,
> >  		}
> >  		ZVAL_DEREF(z);
> >  		SEPARATE_ZVAL_NOREF(z);
> > -		if (inc) {
> > -			increment_function(z);
> > +		if (ZEND_OBJECT_HAS_TYPE_HINTS(object) && (prop_info = zend_object_fetch_property_type_info(object, property, cache_slot))) {
> > +			zval tmp;
> > +
> > +			ZVAL_DUP(&tmp, z);
>
> why do you duplicate here? we even don't exactly know if this property has
> a type hint.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1797/files/46742861d254a5fe74b6d88dde4232e4bae4bb5c#r58685401>
>